tools/docgen/docgen/channels.py

```diff
@@ -41,3 +51,207 @@ def process(self, input_file: str) -> ():
+            print("Error converting the RST file to Markdown:", e)
+            return None
+    
+    def _find_add_extension(self, parsed_html):
+        for link in parsed_html.find_all('a', href=True):
+            href = link['href']
+            if not self._is_valid_url(href) and '.md' not in href:
+                split_href = href.split("#")
+                split_href[0] += ".md"
+                new_href = "#".join(split_href)
+                link['href'] = new_href
+        return parsed_html
+
+    def process(self, input_file: str, index: int) -> ():
+        print(f"Processing {input_file} for fabric")
+        output_file = os.path.join(self.output_dir, input_file)
+        output_img_dir = self.media_dir + "/" + self._sentence_to_snake(input_file)
+        full_input_file = os.path.join(self.input_dir, input_file)
+        notebook_path = self.notebooks[index]['path']
+
+        def callback(el):
+            if el.contents[0].has_attr('class'):
+                return el.contents[0]["class"][0].split("-")[-1] if len(el.contents) >= 1 else None
+            else:
+                return el['class'][0] if el.has_attr('class') else None
+        
+        def convert_soup_to_md(soup, **options):
+            return MarkdownConverter(**options).convert_soup(soup)
+
+        if str(input_file).endswith(".rst"):
+            output_file = self._sentence_to_snake(str(output_file).replace(".rst", ".md"))
+            html = self._read_rst(full_input_file)
+            parsed_html = markdown.markdown(html, extensions=["markdown.extensions.tables", "markdown.extensions.fenced_code"])
+            parsed_html = BeautifulSoup(parsed_html)
+            parsed_html = self._download_and_replace_images(parsed_html, None, output_img_dir, os.path.dirname(output_file), notebook_path, True)
+            parsed_html = self._find_add_extension(parsed_html)
+        
+        elif str(input_file).endswith(".ipynb"):
+            output_file = self._sentence_to_snake(str(output_file).replace(".ipynb", ".md"))
+            parsed = read(full_input_file, as_version=4)
+
+            c = Config()
+            c.TagRemovePreprocessor.remove_cell_tags = (self.hide_tag,)
+            c.TagRemovePreprocessor.enabled = True
+            c.MarkdownExporter.preprocessors = ["nbconvert.preprocessors.TagRemovePreprocessor"]
+            md, resources = MarkdownExporter(config=c).from_notebook_node(parsed)
+
+            html = markdown.markdown(md, extensions=["markdown.extensions.tables", "markdown.extensions.fenced_code"])
+            parsed_html = BeautifulSoup(html)
+            # Download images and place them in media directory while updating their links
+            parsed_html = self._download_and_replace_images(parsed_html, resources, output_img_dir, os.path.dirname(output_file), None, False)
+            # Remove StatementMeta
+            for element in parsed_html.find_all(text=re.compile("StatementMeta\(.*?Available\)")):
```

Comment:
These two can be put in both branches of it statement right, also statement meta check should be pushed upstream to the actual notebooks when possible because we don’t want them there either

Reply:
This is for Sempy doc. We do not have that in our notebooks and they have that line in some of their example. Just want to put it there to have a cleaner output for their samples.

Reply:
yes makes sense, but this would be a no-op for us (And would be helpful if our docs had those by mistake)

Reply:
Adding a warning... do you want to it be an error?